### PR TITLE
MultiMedias: implements the possibility to use shared medias (twinnable)

### PR DESCRIPTION
**Pull request by shaoshiva**
Before this PR it was only possible to handle classical medias :

``` php
'image' => array(
    'label' => '',
    'renderer' => 'Novius\Renderers\Renderer_MultiMedias',
    'renderer_options' => array(
        'key' => 'image',
        'mode' => 'image',
    ),
),
```

Now you can still handle classical medias but also shared media (twinnable) :

``` php
'medias->image' => array(
    'label' => '',
    'renderer' => 'Novius\Renderers\Renderer_MultiMedias',
    'renderer_options' => array(
        'mode' => 'image',
    ),
),
```

``` php
'shared_medias_context->image' => array(
    'label' => '',
    'renderer' => 'Novius\Renderers\Renderer_MultiMedias',
    'renderer_options' => array(
        'mode' => 'image',
    ),
),
```

As you see the renderer option `key` is no longer used, now the key is automatically extracted from the field name. If this option is set then the renderer will work as before the PR (for retro-compatibility).

It is also no longer necessary to write the `before_save` callback on the field, the renderer will handle it automatically (except if the `key` option is set).

### CODE REVIEW

File: classes/renderer/multimedias.php [hunk 1]
```diff
@@ -9,44 +9,150 @@ class Renderer_MultiMedias extends Renderer
...
+    }
+
+    /**
+     * Gets the value from the instance
+     *
+     * @param $item
+     * @param null $key
+     * @return array
+     */
+    protected function getValueFromInstance($item, $key = null)
```
Comment (jguyomard) [inline classes/renderer/multimedias.php]: argument $key is never used, remove it ?

**Comment by jguyomard** [comment]
:+1:

File: classes/renderer/multimedias.php (hunk 1 above)
Comment (shaoshiva) [inline classes/renderer/multimedias.php]: Well seen, I forgot to remove it.

**Comment by SwithFr** [comment]
Deletion don't work :(

**Comment by shaoshiva** [comment]
@SwithFr fixed :)

**Comment by SwithFr** [comment]
Perfect ;)
:+1:

**Pull request merged**